sys/include/event/periodic_callback.h

```diff
@@ -175,6 +175,29 @@ static inline void event_periodic_callback_stop(event_periodic_callback_t *event
     event_periodic_stop(&event->periodic);
 }
 
+/**
+ * @brief   Initialize and start a periodic callback event that will be executed once
+ *
+ * This is a convenience function that combines @ref event_periodic_callback_init
+ * and @ref event_periodic_callback_start
+ *
+ * @param[out]  event           event_periodic_callback object to initialize
+ * @param[in]   queue           queue that the timed-out event will be added to
+ * @param[in]   clock           the clock to configure this timer on
+ * @param[in]   timeout         time after which the event should be executed
+ * @param[in]   callback        callback to set up
+ * @param[in]   arg             callback argument to set up
+ */
+static inline void event_periodic_callback_oneshot(event_periodic_callback_t *event,
+                                                   event_queue_t *queue,
+                                                   ztimer_clock_t *clock, uint32_t timeout,
+                                                   void (*callback)(void *), void *arg)
+{
+    event_periodic_callback_init(event, clock, queue, callback, arg);
```

Comment:
I think a `event_periodic_callback_stop` would be safer before reinit, because you assume that it is not running?
But if it is currently active, touching it by reinitialization could be dangerous

Reply:
this precondition is not mention for for 'event_periodic_callback_init',  but obvious since init zeros the structure -> possibly destroying lists
please add that to that documentation

Reply:
''event_periodic_stop / event_periodic_callback_stop` is not enough  (it only stops the ztimer but doe not remove the event if allredy queued) zeros -> destoyed queue

Reply:
maybe all those fears are not relevant since each event_periodic_callback_t is only to be used (initialised) once

Reply:
> maybe all those fears are not relevant since each event_periodic_callback_t is only to be used (initialised) once

I guess you are right.